src/bin/main.rs

```diff
@@ -342,6 +346,8 @@ fn main() {
         Ok(summary) => {
             if summary.has_operational_errors() {
                 1
```

Comment:
This part is a bit wrong. IIRC in check mode, the exit code should be either 0 or 1, but this change does not seem to respect it.

Reply:
Thanks for the feedback! I haven't gotten around to implementing https://github.com/rust-lang-nursery/rustfmt/issues/1977#issuecomment-372167773, but when I do later today, this the exit code handling _should_ be far nicer and actually correct.